Sources/Amplitude/Amplitude.m

```diff
@@ -880,22 +886,44 @@ - (NSDictionary *)mergeEventsAndIdentifys:(NSMutableArray *)events identifys:(NS
                     ([[event objectForKey:SEQUENCE_NUMBER] longLongValue] <
                      [[identify objectForKey:SEQUENCE_NUMBER] longLongValue])) {
                 [events removeObjectAtIndex:0];
-                maxEventId = [[event objectForKey:EVENT_ID] longValue];
                 identify = nil;
             } else {
                 [identifys removeObjectAtIndex:0];
-                maxIdentifyId = [[identify objectForKey:EVENT_ID] longValue];
                 event = nil;
             }
         }
 
         [mergedEvents addObject:event != nil ? event : identify];
     }
 
-    NSDictionary *results = [[NSDictionary alloc] initWithObjectsAndKeys:mergedEvents, EVENTS, [NSNumber numberWithLongLong:maxEventId], MAX_EVENT_ID, [NSNumber numberWithLongLong:maxIdentifyId], MAX_IDENTIFY_ID, nil];
+    NSDictionary *results = [[NSDictionary alloc] initWithObjectsAndKeys:mergedEvents, EVENTS, nil];
     return results;
 }
 
+- (NSMutableArray *)removeEventFromBuffer:(NSArray *)buffer currentEventString:(NSString *)currentEventString {
+    NSMutableArray *updatedBuffer = [buffer mutableCopy];
+    NSUInteger *index = 0;
+    for (NSDictionary *event in buffer) {
+        NSError *error = nil;
+        NSData *eventsDataLocal = nil;
+        eventsDataLocal = [NSJSONSerialization dataWithJSONObject:event options:0 error:&error];
+        if (error != nil) {
+            AMPLITUDE_ERROR(@"ERROR: NSJSONSerialization error: %@", error);
+            return [buffer mutableCopy];
+        }
+
+        NSString *eventString = [[NSString alloc] initWithData:eventsDataLocal encoding:NSUTF8StringEncoding];
+        //remove the [ and ] in the currentEventString for comparing with event dictionary
+        NSString *currentEventDictionaryString =  [currentEventString substringWithRange:NSMakeRange(1,  currentEventString.length - 2)];
+        if ([eventString isEqualToString:currentEventDictionaryString]) {
+            [updatedBuffer removeObject:event];
```

Comment:
nit: for just delete one item is fine, just this method will do another linear search in the buffer.